src/Preg.php

```diff
@@ -25,8 +25,21 @@
 final class Preg
 {
     /**
-     * @param null|string[]         $matches
-     * @param int-mask<0, 256, 512> $flags
+     * @template TFlags as int-mask<0, 256, 512>
+     *
+     * @param null|mixed[] $matches
+     * @param TFlags       $flags
+     *
+     * @param-out (TFlags is 256
+     *     ? array<array-key, array{string, 0|positive-int}|array{'', -1}>
+     *     : (TFlags is 512
+     *         ? array<array-key, string|null>
+     *             : (TFlags is 768
+     *                 ? array<array-key, array{string, 0|positive-int}|array{null, -1}>
```

Comment:
`''` is not `null`, is it? 😉

Changing either one is resulting in failing PHPStan analyze.

Reply:
I know it's different, I just asked whether it's different on purpose and _actually_ differs (because it seems weird) 🙂. If this is the proper definition of potential types, then fine 👍.